ideascube/management/commands/tags.py

```diff
@@ -121,3 +135,53 @@ def list(self, options):
         print(row.format('.' * 40, '.' * 40, '.' * 40))
         for tag in Tag.objects.order_by('slug'):
             print(row.format(tag.name, tag.slug, self._count(tag.name)))
+
+    def sanitize(self, options):
+        all_tags = list(Tag.objects.all().order_by('id').reverse())
+        while all_tags:
+            tag = all_tags.pop()
+            new_tag_name = sanitize_tag_name(tag.name)
+
+            if not new_tag_name:
+                # No need to delete relation, happy cascading !
+                tag.delete()
+                continue
+
+            other_equivalent_tags = {t for t in all_tags
+                if new_tag_name == sanitize_tag_name(t.name)}
+
+            # All the relations we need to redirect to `tag`
+            other_relations = TaggedItem.objects.filter(
+                tag__in=other_equivalent_tags)
+            for relation in other_relations:
+                # if an object `o` is tagged with tag `foo` and `Foo`, the
+                # relation `o-Foo` must be change to `o-foo`. But this relation
+                # already exists, so, instead, we must delete `o-Foo`,
+                # not change it.
+                try:
+                    TaggedItem.objects.get(
+                        tag=tag,
+                        object_id=relation.content_object.id,
+                        content_type = relation.content_type)
+                except ObjectDoesNotExist:
+                    # We must change the relation
+                    relation.tag = tag
+                    relation.save()
+                except MultipleObjectsReturned:
+                    # This should not happend, we cannot have the *exact* same
+                    # relation twice. But let's be safe.
+                    relation.delete()
+                else:
+                    # We must delete the relation
+                    relation.delete()
+
+            # There is no relation to other tags, delete them.
```

Comment:
This gave me a pause: I wondered why there would be no relation to the other tags, until I realized that there is no relation to other tags **left** at this point (because we deleted or moved them just above).

What a difference a word can make. 😄